presto-main/src/main/java/com/facebook/presto/sql/planner/plan/ExchangeNode.java

```diff
@@ -96,6 +99,7 @@ public ExchangeNode(
         this.scope = scope;
         this.partitioningScheme = partitioningScheme;
         this.inputs = listOfListsCopy(inputs);
+        this.orderSensitive = orderSensitive;
```

Comment:
I would extract adding orderSensitive fields to this node and `RemoteSourceNode` as separate commit with a `checkState` in `LocalExecutionPlanner` that it is not set. Then as a subsequent commit I would add support for order sensitive distribution in execution. That way this commit will be not polluted and it will be easier to review, merge and maintain etc.